src/mlpack/methods/linear_svm/linear_svm.hpp

```diff
@@ -290,9 +275,9 @@ class LinearSVM
   ElemType Train(const MatType& data,
                  const arma::Row<size_t>& labels,
                  const size_t numClasses,
-                 const double lambda,
-                 const double delta,
-                 const bool fitIntercept,
+                 const std::optional<double> lambda = std::nullopt,
```

Comment:
I looked through the failing build.  I see that this overload, and also the one that just takes callbacks and no hyperparameters, both look the same to the compiler.  (It compiles successfully on my machine, probably different versions of compilers treat it differently.)

Anyway, the intent is that a user can do either of the following:

```
Train(data, labels, numClasses, callbacks...)
Train(data, labels, numClasses, lambda)
```

but if the user calls `Train(data, labels, numClasses)`, we can just use the first version with empty callbacks... so we can change this line here to

```suggestion
                 const double lambda,
```